src/main.rs

```diff
@@ -198,23 +207,37 @@ fn check_once(
 }
 
 fn dump_stacktrace(trace: &StackTrace) {
-    eprintln!(
-        "    Thread {:x} ({}{})",
-        trace.thread_id,
-        trace.status_str(),
-        if trace.owns_gil { ", holding GIL" } else { "" }
+    log_json(
+        Severity::Info,
+        &format!(
+            "    Thread {:x} ({}{})",
+            trace.thread_id,
+            trace.status_str(),
+            if trace.owns_gil { ", holding GIL" } else { "" }
+        ),
+        None,
     );
+
     for frame in trace.frames.iter().rev() {
```

Comment:
Can we log the entire stack into a single json blob?

Reply:
I think we should retain the legacy logging as-is and allow you to pass in a config opt that dumps JSON instead of wholesale moving the library over. This is a breaking change if others are relying on the current format.